src/States/MapViewStateDraw.cpp

```diff
@@ -69,49 +69,60 @@ void MapViewState::drawMiniMap()
 	bool isHeightmapToggled = mBtnToggleHeightmap.toggled();
 	renderer.drawImage(isHeightmapToggled ? mHeightMap : mMapDisplay, miniMapBoxFloat.startPoint());
 
-	if (ccLocationX() != 0 && ccLocationY() != 0)
+	const auto miniMapOffset = mMiniMapBoundingBox.startPoint() - NAS2D::Point<int>{0, 0};
```

Comment:
`x - 0` is still `x`.

Reply:
It's a type conversion. `Point - Point = Vector`. I needed a `Vector`. That was the easiest way I could think of to convert.

Reply:
This is why the Mathematician definition of a point can be defined as "A vector with no direction nor magnitude" and why I was advocating in the beginning that the distinction should not be made as two separate classes. With the appropriate variable names just using `Vector` when referring to points and `Vector - Vector` when referring to directions removes confusions like this and is a 30-year-old convention. Don't try to buck the trend now. You're just going to confuse people who know what they're doing.

Reply:
I think this may be a symptom of the design lacking clarification, or perhaps a weakness in the API being offered. Personally I like being able to get errors when I try to do something stupid like adding two `Point`s together.

In this case, there isn't much clarity on what the fixed basis is, and what is a relative offset. There's a bit of ambiguity here, in that you could go either way. Screen coordinates might be considered absolute, while map positions are relative, or map positions are absolute, while where things appear on the screen are relative. I'm inclined to go more with the later.

Keep in mind this code was originally written before there was a `Point`/`Vector` distinction.

----

In terms of improving the API, I think it would be good if the `MiniMap` could be rendered using it's own viewport, with it's own origin, and size matching the map. That would be a way to draw the minimap without having to manually offset every drawing call.

Anyway, that's an issue for another day.

Reply:
At the very least, an off-screen memory texture that is then drawn in the correct place.